File: src/Java/myservicebus-amazon-sqs/src/main/java/com/myservicebus/amazon/sqs/AmazonSqsTransportException.java
package com.myservicebus.amazon.sqs;

public final class AmazonSqsTransportException extends RuntimeException {
    private final String operation;
    private final String entityName;

    public AmazonSqsTransportException(String operation, String entityName, Throwable cause) {
        super("Amazon SQS/SNS could not " + operation + " for entity '" + entityName + "'.", cause);
        this.operation = operation;
        this.entityName = entityName;
    }

    public String getOperation() {
        return operation;
    }

    public String getEntityName() {
        return entityName;
    }
}

File: src/MyServiceBus.AmazonSqs/AmazonSqsTransportException.cs
namespace MyServiceBus;

public sealed class AmazonSqsTransportException : Exception
{
    public AmazonSqsTransportException(string operation, string entityName, Exception innerException)
        : base($"Amazon SQS/SNS could not {operation} for entity '{entityName}'.", innerException)
    {
        Operation = operation;
        EntityName = entityName;
    }

    public string Operation { get; }
    public string EntityName { get; }
}
